src/aws/ecr_helpers.go

```diff
@@ -0,0 +1,127 @@
+package aws
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/Originate/exosphere/src/config"
+	"github.com/Originate/exosphere/src/docker"
+	"github.com/Originate/exosphere/src/types"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
+	"github.com/moby/moby/client"
+)
+
+// PushImages does the following:
+// Create the image repository on ECR if it doesn't exist
+// Tags application images
+// Retrieves ECR credentials
+// Pushes images to ECR
+// Returns a map from service name to image name on ECR
+func PushImages(deployConfig types.DeployConfig, dockerComposePath string) (map[string]string, error) {
+	config := aws.NewConfig().WithRegion(deployConfig.AwsConfig.Region)
+	session := session.Must(session.NewSession())
+	ecrClient := ecr.New(session, config)
+	dockerClient, err := client.NewEnvClient()
+	if err != nil {
+		return nil, err
+	}
+	dockerCompose, err := docker.GetDockerCompose(dockerComposePath)
+	if err != nil {
+		return nil, err
+	}
+	imagesMap, err := getImageNames(deployConfig, filepath.Dir(dockerComposePath), dockerCompose)
+	if err != nil {
+		return nil, err
+	}
+	for serviceName, imageName := range imagesMap {
+		deployConfig.LogChannel <- fmt.Sprintf("Pushing image for: %s...\n\n", serviceName)
+		repositoryName, version := getRepositoryConfig(deployConfig, imageName)
+		repositoryURI, err := createRepository(ecrClient, repositoryName)
+		if err != nil {
+			return nil, err
+		}
+		taggedImage := fmt.Sprintf("%s:%s", repositoryURI, version)
+		err = docker.TagImage(dockerClient, imageName, taggedImage)
+		if err != nil {
+			return nil, err
+		}
+		imagesMap[serviceName] = taggedImage
+		encodedAuth, err := getECRCredentials(ecrClient)
+		if err != nil {
+			return nil, err
+		}
+		err = docker.PushImage(dockerClient, taggedImage, encodedAuth)
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Can this function be broken up a bit more. Maybe extract whats in the for loop into its own function?